token-client/src/test/java/com/sap/cloud/security/xsuaa/client/XsuaaDefaultEndpointsTest.java

```diff
@@ -5,32 +5,84 @@
  */
 package com.sap.cloud.security.xsuaa.client;
 
+import com.sap.cloud.security.config.CredentialType;
+import com.sap.cloud.security.config.OAuth2ServiceConfiguration;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+import java.net.URI;
+
 import static org.hamcrest.CoreMatchers.is;
 import static org.hamcrest.MatcherAssert.assertThat;
 
-import org.junit.Test;
-
 public class XsuaaDefaultEndpointsTest {
 
+	private static OAuth2ServiceConfiguration oAuth2ServiceConfiguration;
+	private static final String URL = "https://subdomain.myauth.com";
+	private static final String CERT_URL = "https://subdomain.cert.myauth.com";
+
+	@Before
+	public void setUp() {
+		oAuth2ServiceConfiguration = Mockito.mock(OAuth2ServiceConfiguration.class);
+		Mockito.when(oAuth2ServiceConfiguration.getUrl()).thenReturn(URI.create(URL));
+	}
+
 	@Test
 	public void getTokenEndpoint() {
-		OAuth2ServiceEndpointsProvider cut = createXsuaaDefaultEndpointProvider("https://subdomain.myauth.com");
+		OAuth2ServiceEndpointsProvider cut = createXsuaaDefaultEndpointProvider(URL);
+
+		assertThat(cut.getTokenEndpoint().toString(), is(URL + "/oauth/token"));
 
-		assertThat(cut.getTokenEndpoint().toString(), is("https://subdomain.myauth.com/oauth/token"));
+		Mockito.when(oAuth2ServiceConfiguration.getCredentialType()).thenReturn(CredentialType.INSTANCE_SECRET);
+		OAuth2ServiceEndpointsProvider cut2 = new XsuaaDefaultEndpoints(oAuth2ServiceConfiguration);
+
+		assertThat(cut2.getTokenEndpoint().toString(), is(URL + "/oauth/token"));
+	}
+
+	@Test
+	public void getTokenEndpointX509() {
+		Mockito.when(oAuth2ServiceConfiguration.getCertUrl()).thenReturn(URI.create(CERT_URL));
+		Mockito.when(oAuth2ServiceConfiguration.getCredentialType()).thenReturn(CredentialType.X509);
+		OAuth2ServiceEndpointsProvider cut = new XsuaaDefaultEndpoints(oAuth2ServiceConfiguration);
+		assertThat(cut.getTokenEndpoint().toString(), is(CERT_URL + "/oauth/token"));
```

Comment:
assertion must be `is(URL + "/oauth/token")`

Reply:
@nenaraab, nope, it won't work if credential-type is x509, then also for token retrieval mTLS needs to be used, meaning the cert url needs to be called, otherwise it'll return 401 
```json
{
    "error": "unauthorized",
    "error_description": "Unsupported instance credential for ..."
}
```